personalitytest/static/js/src/personalitytest-view.js

```diff
@@ -32,50 +69,36 @@ function PersonalityTestXBlockStudent(runtime, element) {
                 $.post(answersHandleUrl, '{}')
                     .done(function (resp) {
                         var questions = JSON.parse(response['questions']);
+                        $('.personality-test-description', element).append(paragraphize(response['quizz_description']));
                         var studentAnswers = '';
                         if (resp.success) {
                             studentAnswers = JSON.parse(resp['answers']);
                         }
-
                         var myForm = $('.personality-test-form', element);
                         var mainDiv = $('.personality-test-form-table', element);
 
-                        var list = document.createElement('ol');
+                        var list = $('<ol />');
                         var i = 0;
+                        var ii = 0;
                         questions.forEach(function (question) {
-                            var questionInList = document.createElement('li');
-                            var spanQuestion = document.createElement('div');
-                            spanQuestion.className = 'question-div';
-
-                            var select = document.createElement('select');
-                            var opt = document.createElement('option');
-                            opt.text = '';
-                            opt.value = '';
-                            select.add(opt);
-                            var studentAnswer = '';
-                            if (studentAnswers !== '') {
-                                studentAnswer = getAnswersValue(studentAnswers, question['id']);
-                            }
-                            question['answers'].forEach(function (answer) {
-                                var option = document.createElement('option');
-                                option.text = answer.answer;
-                                option.value = question['id'];
-                                if (studentAnswers !== '' && studentAnswer === answer.answer) {
-                                    option.selected = true;
-                                }
-                                select.add(option);
-                            });
-
-                            spanQuestion.appendChild(document.createTextNode(++i + '. ' + question.description));
-                            questionInList.appendChild(spanQuestion);
-
-                            var spanAnswer = document.createElement('div');
-                            spanAnswer.className = 'answer-div';
-                            spanAnswer.appendChild(select);
-
-                            questionInList.appendChild(spanAnswer);
-                            list.appendChild(questionInList);
+                            ++i;
```

Comment:
what’s the point of this `i` variable ?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

The callback gives you the index.